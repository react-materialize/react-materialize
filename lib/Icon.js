'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _constants = require('./constants');

var _constants2 = _interopRequireDefault(_constants);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Icon = function (_React$Component) {
  _inherits(Icon, _React$Component);

  function Icon() {
    _classCallCheck(this, Icon);

    return _possibleConstructorReturn(this, (Icon.__proto__ || Object.getPrototypeOf(Icon)).apply(this, arguments));
  }

  _createClass(Icon, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var classes = {
        'material-icons': true
      };
      _constants2.default.PLACEMENTS.forEach(function (p) {
        classes[p] = _this2.props[p];
      });
      _constants2.default.ICON_SIZES.forEach(function (s) {
        classes[s] = _this2.props[s];
      });
      return _react2.default.createElement(
        'i',
        { className: (0, _classnames2.default)(classes, this.props.className) },
        this.props.children
      );
    }
  }]);

  return Icon;
}(_react2.default.Component);

Icon.propTypes = {
  /*
   * Classname passed to i tag
   */
  className: _propTypes2.default.string,
  /*
   * Icon type: <a href='https://material.io/icons/'>https://material.io/icons/</a>
   */
  children: _propTypes2.default.string,
  /*
   * Placement for icon if used beside a text ↓
   */
  left: _propTypes2.default.bool,
  center: _propTypes2.default.bool,
  right: _propTypes2.default.bool,
  /*
   * Sizes for icons ↓
   */
  tiny: _propTypes2.default.bool,
  small: _propTypes2.default.bool,
  medium: _propTypes2.default.bool,
  large: _propTypes2.default.bool
};

exports.default = Icon;