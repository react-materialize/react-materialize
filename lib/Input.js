'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _idgen = require('./idgen');

var _idgen2 = _interopRequireDefault(_idgen);

var _constants = require('./constants');

var _constants2 = _interopRequireDefault(_constants);

var _Icon = require('./Icon');

var _Icon2 = _interopRequireDefault(_Icon);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Input = function (_Component) {
  _inherits(Input, _Component);

  function Input(props) {
    _classCallCheck(this, Input);

    var _this = _possibleConstructorReturn(this, (Input.__proto__ || Object.getPrototypeOf(Input)).call(this, props));

    _this.state = {
      value: props.value || props.defaultValue,
      checked: !!props.checked
    };

    _this._onChange = _this._onChange.bind(_this);
    _this.getMultipleValues = _this.getMultipleValues.bind(_this);
    _this.isSelect = _this.isSelect.bind(_this);
    return _this;
  }

  _createClass(Input, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      if (this.isMaterialSelect()) {
        $(this.selectInput).material_select();
        $(this.selectInput).on('change', this._onChange);
      }
      if (this.isDatePicker) {
        $(this.dateInput).pickadate(this.props.options);
        $(this.dateInput).on('change', this._onChange);
      }
      if (this.isTimePicker) {
        $(this.timeInput).pickatime(this.props.options);
        $(this.timeInput).on('change', this._onChange);
      }
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate() {
      if (this.isMaterialSelect() && !this.props.multiple) {
        $(this.selectInput).material_select();
      }
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      var _this2 = this;

      if (this.isMaterialSelect() && !this.props.multiple) {
        this.setState({
          value: nextProps.defaultValue
        }, function () {
          return $(_this2.selectInput).material_select();
        });
      }
    }
  }, {
    key: 'componentWillUnmount',
    value: function componentWillUnmount() {
      if (this.isMaterialSelect()) {
        $(this.selectInput).off('change', this._onChange);
      }
    }
  }, {
    key: 'getMultipleValues',
    value: function getMultipleValues(_ref) {
      var options = _ref.options;

      if (!options) {
        return;
      }

      return Array.from(options).filter(function (opt) {
        return opt.selected;
      }).map(function (opt) {
        return opt.value;
      });
    }
  }, {
    key: '_onChange',
    value: function _onChange(e) {
      var onChange = this.props.onChange;

      var types = {
        checkbox: e.target.checked,
        radio: e.target.checked,
        'select-multiple': this.getMultipleValues(e.target),
        default: e.target.value
      };
      var value = types.hasOwnProperty(e.target.type) ? types[e.target.type] : types['default'];
      if (onChange) {
        onChange(e, value);
      }

      this.setState({ value: value, checked: e.target.checked });
    }
  }, {
    key: 'render',
    value: function render() {
      var _this3 = this;

      var _props = this.props,
          browserDefault = _props.browserDefault,
          children = _props.children,
          className = _props.className,
          labelClassName = _props.labelClassName,
          defaultValue = _props.defaultValue,
          error = _props.error,
          label = _props.label,
          multiple = _props.multiple,
          placeholder = _props.placeholder,
          success = _props.success,
          s = _props.s,
          m = _props.m,
          l = _props.l,
          type = _props.type,
          validate = _props.validate,
          onLabel = _props.onLabel,
          offLabel = _props.offLabel,
          inline = _props.inline,
          other = _objectWithoutProperties(_props, ['browserDefault', 'children', 'className', 'labelClassName', 'defaultValue', 'error', 'label', 'multiple', 'placeholder', 'success', 's', 'm', 'l', 'type', 'validate', 'onLabel', 'offLabel', 'inline']);

      var sizes = { s: s, m: m, l: l };
      this._id = this._id || this.props.id || 'input_' + (0, _idgen2.default)();
      var classes = {
        col: true,
        inline: type !== 'checkbox' && type !== 'radio' && inline,
        'input-field': type !== 'checkbox' && type !== 'radio' && type !== 'file',
        'file-field': type === 'file'
      };
      _constants2.default.SIZES.forEach(function (size) {
        classes[size + sizes[size]] = sizes[size];
      });
      var inputClasses = {
        validate: validate,
        invalid: error,
        valid: success,
        'browser-default': browserDefault && this.isSelect()
      };
      var C = void 0,
          inputType = void 0;
      switch (type) {
        case 'textarea':
          C = 'textarea';
          inputClasses['materialize-textarea'] = true;
          break;
        case 'switch':
          C = 'input';
          inputType = 'checkbox';
          break;
        case 'file':
          C = 'input';
          inputClasses['file-path'] = true;
          break;
        default:
          C = 'input';
          inputType = type || 'text';
      }
      var labelClasses = {
        active: this.state.value || this.isSelect() || placeholder
      };

      var htmlLabel = label || inputType === 'radio' ? _react2.default.createElement(
        'label',
        {
          className: (0, _classnames2.default)(labelClasses, labelClassName),
          'data-success': success,
          'data-error': error,
          htmlFor: this._id
        },
        label
      ) : null;

      if (this.isSelect()) {
        var options = placeholder && !defaultValue ? [_react2.default.createElement(
          'option',
          { disabled: true, key: (0, _idgen2.default)() },
          placeholder
        )] : [];
        options = options.concat(_react2.default.Children.map(children, function (child) {
          return _react2.default.cloneElement(child, { key: child.props.value });
        }));

        return _react2.default.createElement(
          'div',
          { className: (0, _classnames2.default)(classes) },
          this.renderIcon(),
          htmlLabel,
          _react2.default.createElement(
            'select',
            _extends({}, other, {
              multiple: multiple,
              id: this._id,
              className: (0, _classnames2.default)(className, inputClasses),
              ref: function ref(_ref2) {
                return _this3.selectInput = _ref2;
              },
              defaultValue: defaultValue
            }),
            options
          )
        );
      } else if (type === 'date') {
        this.isDatePicker = true;
        delete other.options;

        return _react2.default.createElement(
          'div',
          { className: (0, _classnames2.default)(classes) },
          this.renderIcon(),
          _react2.default.createElement(C, _extends({}, other, {
            className: (0, _classnames2.default)(className, inputClasses),
            defaultValue: defaultValue,
            id: this._id,
            ref: function ref(_ref3) {
              return _this3.dateInput = _ref3;
            },
            placeholder: placeholder,
            type: 'date'
          })),
          htmlLabel
        );
      } else if (type === 'time') {
        this.isTimePicker = true;
        delete other.options;

        return _react2.default.createElement(
          'div',
          { className: (0, _classnames2.default)(classes) },
          this.renderIcon(),
          _react2.default.createElement(C, _extends({}, other, {
            className: (0, _classnames2.default)(className, inputClasses),
            defaultValue: defaultValue,
            id: this._id,
            ref: function ref(_ref4) {
              return _this3.timeInput = _ref4;
            },
            placeholder: placeholder
          })),
          htmlLabel
        );
      } else if (type === 'switch') {
        return _react2.default.createElement(
          'div',
          { className: 'switch' },
          _react2.default.createElement(
            'label',
            null,
            offLabel || 'Off',
            _react2.default.createElement('input', _extends({}, other, { onChange: this._onChange, type: 'checkbox' })),
            _react2.default.createElement('span', { className: 'lever' }),
            onLabel || 'On'
          )
        );
      } else if (type === 'file') {
        return _react2.default.createElement(
          'div',
          { className: (0, _classnames2.default)(classes) },
          _react2.default.createElement(
            'div',
            { className: 'btn' },
            _react2.default.createElement(
              'span',
              null,
              label
            ),
            _react2.default.createElement(C, _extends({ type: 'file', multiple: multiple }, other))
          ),
          _react2.default.createElement(
            'div',
            { className: 'file-path-wrapper' },
            _react2.default.createElement(C, {
              type: 'text',
              className: (0, _classnames2.default)(className, inputClasses),
              defaultValue: defaultValue,
              id: this._id,
              placeholder: placeholder
            })
          )
        );
      } else {
        var _defaultValue = inputType !== 'checkbox' && inputType !== 'radio' ? this.state.value : _defaultValue;

        if (inputType === 'checkbox' || inputType === 'radio') {
          return _react2.default.createElement(
            'div',
            { className: (0, _classnames2.default)(classes) },
            this.renderIcon(),
            _react2.default.createElement(C, _extends({}, other, {
              className: (0, _classnames2.default)(className, inputClasses),
              ref: function ref(_ref5) {
                return _this3.input = _ref5;
              },
              id: this._id,
              checked: this.state.checked,
              onChange: this._onChange,
              placeholder: placeholder,
              type: inputType
            })),
            htmlLabel
          );
        }

        return _react2.default.createElement(
          'div',
          { className: (0, _classnames2.default)(classes) },
          this.renderIcon(),
          _react2.default.createElement(C, _extends({}, other, {
            className: (0, _classnames2.default)(className, inputClasses),
            ref: function ref(_ref6) {
              return _this3.input = _ref6;
            },
            defaultValue: _defaultValue,
            id: this._id,
            onChange: this._onChange,
            placeholder: placeholder,
            type: inputType
          })),
          htmlLabel
        );
      }
    }
  }, {
    key: 'renderIcon',
    value: function renderIcon() {
      var _props2 = this.props,
          icon = _props2.icon,
          children = _props2.children;

      if (icon) {
        return _react2.default.createElement(
          _Icon2.default,
          { className: 'prefix' },
          icon
        );
      } else {
        var _icon = null;
        if (_react2.default.Children.count(children) === 1 && !Array.isArray(children)) {
          _icon = _react2.default.Children.only(children);
        }
        return _icon === null ? null : _react2.default.cloneElement(_icon, { className: 'prefix' });
      }
    }
  }, {
    key: 'isSelect',
    value: function isSelect() {
      return this.props.type === 'select';
    }
  }, {
    key: 'isMaterialSelect',
    value: function isMaterialSelect() {
      return this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined';
    }
  }]);

  return Input;
}(_react.Component);

Input.propTypes = {
  s: _propTypes2.default.number,
  m: _propTypes2.default.number,
  l: _propTypes2.default.number,
  inline: _propTypes2.default.bool,
  children: _propTypes2.default.node,
  className: _propTypes2.default.string,
  labelClassName: _propTypes2.default.string,
  label: _propTypes2.default.node,
  error: _propTypes2.default.string,
  success: _propTypes2.default.string,
  /**
   * Input field type, e.g. select, checkbox, radio, text, tel, email, file
   * @default 'text'
   */
  type: _propTypes2.default.string,
  defaultValue: _propTypes2.default.any,
  placeholder: _propTypes2.default.string,
  id: _propTypes2.default.string,
  icon: _propTypes2.default.string,
  name: _propTypes2.default.string,
  validate: _propTypes2.default.bool,
  multiple: _propTypes2.default.bool,
  browserDefault: _propTypes2.default.bool,
  checked: _propTypes2.default.bool,
  onLabel: _propTypes2.default.string,
  offLabel: _propTypes2.default.string,
  onChange: _propTypes2.default.func,
  options: _propTypes2.default.any,
  /**
   * Value used to set a initial value
   */
  value: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.array])
};

Input.defaultProps = { type: 'text', checked: false };

exports.default = Input;