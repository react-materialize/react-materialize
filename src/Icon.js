import React from 'react';
import PropTypes from 'prop-types';
import constants from './constants';
import cx from 'classnames';

class Icon extends React.Component {
  render() {
    const classes = {
      'material-icons': true
    };
    constants.PLACEMENTS.forEach(p => {
      classes[p] = this.props[p];
    });
    constants.ICON_SIZES.forEach(s => {
      classes[s] = this.props[s];
    });
    return (
      <i className={cx(classes, this.props.className)}>{this.props.children}</i>
    );
  }
}

Icon.propTypes = {
  /*
   * Classname passed to i tag
   */
  className: PropTypes.string,
  /*
   * Icon type: <a href='https://material.io/icons/'>https://material.io/icons/</a>
   */
  children: PropTypes.string,
  /*
   * Placement for icon if used beside a text ↓
   */
  left: PropTypes.bool,
  center: PropTypes.bool,
  right: PropTypes.bool,
  /*
   * Sizes for icons ↓
   */
  tiny: PropTypes.bool,
  small: PropTypes.bool,
  medium: PropTypes.bool,
  large: PropTypes.bool
};

export default Icon;
