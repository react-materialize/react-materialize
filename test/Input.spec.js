import React from 'react';
import { shallow, mount } from 'enzyme';
import Input from '../src/Input';
import Icon from '../src/Icon';
import mocker from './helper/mocker';

describe('<Input />', () => {
  let wrapper;
  const pickadateMock = jest.fn();
  const pickatimeMock = jest.fn();
  const selectMock = jest.fn();

  const pickadateRestore = mocker('pickadate', pickadateMock);
  const pickatimeRestore = mocker('pickatime', pickatimeMock);
  const selectRestore = mocker('material_select', selectMock);

  afterAll(() => {
    pickadateRestore();
    pickatimeRestore();
    selectRestore();
  });

  test('renders', () => {
    wrapper = shallow(<Input />);
    expect(wrapper).toMatchSnapshot();
  });

  test('text input', () => {
    wrapper = shallow(<Input className="red" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('input value', () => {
    wrapper = shallow(<Input value="name" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('label', () => {
    wrapper = shallow(<Input label="Name" labelClassName="red" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('textarea', () => {
    wrapper = shallow(<Input type="textarea" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('select', () => {
    const values = [1, 2, 3, 4, 5];
    wrapper = shallow(
      <Input type="select">
        {values.map(val => <option key={val} value={val} />)}
      </Input>
    );
    expect(wrapper.instance().isSelect()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  test('renders an icon child with prefix class', () => {
    wrapper = shallow(
      <Input type="select" icon="weekend">
        <option value="yes" />
      </Input>
    );
    expect(wrapper.find(Icon).hasClass('prefix')).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  test('placeholder', () => {
    wrapper = shallow(
      <Input type="select" placeholder="Choose your option">
        <option value="1">Option 1</option>
        <option value="2">Option 2</option>
        <option value="3">Option 3</option>
      </Input>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('multiple select', () => {
    wrapper = shallow(
      <Input type="select" multiple placeholder="Choose your option">
        <option value="1">Option 1</option>
        <option value="2">Option 2</option>
        <option value="3">Option 3</option>
      </Input>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('switch', () => {
    const offLabel = 'No';
    const onLabel = 'Yes';
    wrapper = shallow(
      <Input type="switch" onLabel={onLabel} offLabel={offLabel} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('label').text()).toContain(onLabel);
    expect(wrapper.find('label').text()).toContain(offLabel);
  });

  test('radio', () => {
    wrapper = shallow(
      <Input name="group1" type="radio" value="yellow" label="Yellow" />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find({ checked: false })).toHaveLength(1);
  });

  test('radio checked', () => {
    wrapper = shallow(
      <Input
        name="group1"
        type="radio"
        value="yellow"
        label="Yellow"
        checked={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find({ checked: false })).toHaveLength(1);
  });

  test('radio event', () => {
    wrapper = shallow(
      <Input name="group1" type="radio" value="yellow" label="Yellow" />
    );

    expect(wrapper).toMatchSnapshot();
    wrapper
      .find('input')
      .simulate('change', { target: { type: 'radio', checked: true } });

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find({ checked: true })).toHaveLength(1);
  });

  test('radio onChange', () => {
    const onChange = jest.fn();

    wrapper = shallow(
      <Input
        name="group1"
        type="radio"
        value="yellow"
        checked
        label="Yellow"
        onChange={onChange}
      />
    );

    const event = { target: { type: 'radio', checked: false } };
    expect(wrapper).toMatchSnapshot();
    wrapper.find('input').simulate('change', event);
    expect(wrapper).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledWith(event, false);
  });

  test('pickadate', () => {
    const options = { one: 'two' };
    wrapper = shallow(<Input type="date" options={options} />);

    expect(pickadateMock).toHaveBeenCalledWith(options);
    expect(wrapper).toMatchSnapshot();
  });

  test('pickadate with icon', () => {
    wrapper = shallow(
      <Input type="date">
        <Icon>today</Icon>
      </Input>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('pickatime', () => {
    const options = { one: 'two' };
    wrapper = mount(<Input type="time" options={options} />);

    expect(pickatimeMock).toHaveBeenCalledWith(options);
    expect(wrapper).toMatchSnapshot();
  });

  test('file input', () => {
    wrapper = shallow(<Input type="file" label="Upload" multiple />);
    expect(wrapper).toMatchSnapshot();
  });
});
