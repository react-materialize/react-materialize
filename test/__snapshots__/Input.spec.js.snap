// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> file input 1`] = `
<div
  className="col file-field"
>
  <div
    className="btn"
  >
    <span>
      Upload
    </span>
    <input
      checked={false}
      multiple={true}
      type="file"
    />
  </div>
  <div
    className="file-path-wrapper"
  >
    <input
      className="file-path"
      id="input_19"
      type="text"
    />
  </div>
</div>
`;

exports[`<Input /> input value 1`] = `
<div
  className="col input-field"
>
  <input
    checked={false}
    className=""
    defaultValue="name"
    id="input_2"
    onChange={[Function]}
    type="text"
    value="name"
  />
</div>
`;

exports[`<Input /> label 1`] = `
<div
  className="col input-field"
>
  <input
    checked={false}
    className=""
    id="input_3"
    onChange={[Function]}
    type="text"
  />
  <label
    className="red"
    htmlFor="input_3"
  >
    Name
  </label>
</div>
`;

exports[`<Input /> multiple select 1`] = `
<div
  className="col input-field"
>
  <select
    checked={false}
    className=""
    id="input_9"
    multiple={true}
  >
    <option
      disabled={true}
      key="10"
    >
      Choose your option
    </option>
    <option
      key="1/.0"
      value="1"
    >
      Option 1
    </option>
    <option
      key="2/.1"
      value="2"
    >
      Option 2
    </option>
    <option
      key="3/.2"
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`<Input /> pickadate 1`] = `
<div
  className="col input-field"
>
  <input
    checked={false}
    className=""
    id="input_16"
    type="date"
  />
</div>
`;

exports[`<Input /> pickadate with icon 1`] = `
<div
  className="col input-field"
>
  <Icon
    className="prefix"
  >
    today
  </Icon>
  <input
    checked={false}
    className=""
    id="input_17"
    type="date"
  />
</div>
`;

exports[`<Input /> pickatime 1`] = `
<Input
  checked={false}
  options={
    Object {
      "one": "two",
    }
  }
  type="time"
>
  <div
    className="col input-field"
  >
    <input
      checked={false}
      className=""
      id="input_18"
    />
  </div>
</Input>
`;

exports[`<Input /> placeholder 1`] = `
<div
  className="col input-field"
>
  <select
    checked={false}
    className=""
    id="input_7"
  >
    <option
      disabled={true}
      key="8"
    >
      Choose your option
    </option>
    <option
      key="1/.0"
      value="1"
    >
      Option 1
    </option>
    <option
      key="2/.1"
      value="2"
    >
      Option 2
    </option>
    <option
      key="3/.2"
      value="3"
    >
      Option 3
    </option>
  </select>
</div>
`;

exports[`<Input /> radio 1`] = `
<div
  className="col"
>
  <input
    checked={false}
    className=""
    id="input_12"
    name="group1"
    onChange={[Function]}
    type="radio"
    value="yellow"
  />
  <label
    className="active"
    htmlFor="input_12"
  >
    Yellow
  </label>
</div>
`;

exports[`<Input /> radio checked 1`] = `
<div
  className="col"
>
  <input
    checked={false}
    className=""
    id="input_13"
    name="group1"
    onChange={[Function]}
    type="radio"
    value="yellow"
  />
  <label
    className="active"
    htmlFor="input_13"
  >
    Yellow
  </label>
</div>
`;

exports[`<Input /> radio event 1`] = `
<div
  className="col"
>
  <input
    checked={false}
    className=""
    id="input_14"
    name="group1"
    onChange={[Function]}
    type="radio"
    value="yellow"
  />
  <label
    className="active"
    htmlFor="input_14"
  >
    Yellow
  </label>
</div>
`;

exports[`<Input /> radio event 2`] = `
<div
  className="col"
>
  <input
    checked={true}
    className=""
    id="input_14"
    name="group1"
    onChange={[Function]}
    type="radio"
    value="yellow"
  />
  <label
    className="active"
    htmlFor="input_14"
  >
    Yellow
  </label>
</div>
`;

exports[`<Input /> radio onChange 1`] = `
<div
  className="col"
>
  <input
    checked={true}
    className=""
    id="input_15"
    name="group1"
    onChange={[Function]}
    type="radio"
    value="yellow"
  />
  <label
    className="active"
    htmlFor="input_15"
  >
    Yellow
  </label>
</div>
`;

exports[`<Input /> radio onChange 2`] = `
<div
  className="col"
>
  <input
    checked={false}
    className=""
    id="input_15"
    name="group1"
    onChange={[Function]}
    type="radio"
    value="yellow"
  />
  <label
    className=""
    htmlFor="input_15"
  >
    Yellow
  </label>
</div>
`;

exports[`<Input /> renders 1`] = `
<div
  className="col input-field"
>
  <input
    checked={false}
    className=""
    id="input_0"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<Input /> renders an icon child with prefix class 1`] = `
<div
  className="col input-field"
>
  <Icon
    className="prefix"
  >
    weekend
  </Icon>
  <select
    checked={false}
    className=""
    icon="weekend"
    id="input_6"
  >
    <option
      key="yes/.0"
      value="yes"
    />
  </select>
</div>
`;

exports[`<Input /> select 1`] = `
<div
  className="col input-field"
>
  <select
    checked={false}
    className=""
    id="input_5"
  >
    <option
      key=".$1"
      value={1}
    />
    <option
      key=".$2"
      value={2}
    />
    <option
      key=".$3"
      value={3}
    />
    <option
      key=".$4"
      value={4}
    />
    <option
      key=".$5"
      value={5}
    />
  </select>
</div>
`;

exports[`<Input /> switch 1`] = `
<div
  className="switch"
>
  <label>
    No
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="lever"
    />
    Yes
  </label>
</div>
`;

exports[`<Input /> text input 1`] = `
<div
  className="col input-field"
>
  <input
    checked={false}
    className="red"
    id="input_1"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<Input /> textarea 1`] = `
<div
  className="col input-field"
>
  <textarea
    checked={false}
    className="materialize-textarea"
    id="input_4"
    onChange={[Function]}
  />
</div>
`;
